modules/core/02-client/keeper/keeper_test.go

```diff
@@ -483,3 +486,114 @@ func (suite *KeeperTestSuite) TestUnsetParams() {
+			func() {
+				plan.Height = 0
+				expError = sdkerrors.ErrInvalidRequest
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest()  // reset
+			oldPlan.Height = 0 // reset
+
+			path := ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.SetupClients(path)
+			upgradedClientState = suite.chainA.GetClientState(path.EndpointA.ClientID).ZeroCustomFields().(*ibctm.ClientState)
+
+			// use height 1000 to distinguish from old plan
+			plan = upgradetypes.Plan{
+				Name:   "upgrade IBC clients",
+				Height: 1000,
+			}
+
+			tc.malleate()
+
+			// set the old plan if it is not empty
+			if oldPlan.Height != 0 {
+				// set upgrade plan in the upgrade store
+				store := suite.chainA.GetContext().KVStore(suite.chainA.GetSimApp().GetKey(upgradetypes.StoreKey))
+				bz := suite.chainA.App.AppCodec().MustMarshal(&oldPlan)
+				store.Set(upgradetypes.PlanKey(), bz)
+
+				bz, err := types.MarshalClientState(suite.chainA.App.AppCodec(), upgradedClientState)
+				suite.Require().NoError(err)
+
+				suite.chainA.GetSimApp().UpgradeKeeper.SetUpgradedClient(suite.chainA.GetContext(), oldPlan.Height, bz) //nolint:errcheck
+			}
+
+			err := suite.chainA.App.GetIBCKeeper().ClientKeeper.ScheduleIBCClientUpgrade(suite.chainA.GetContext(), plan, upgradedClientState)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+
+				// check that the correct plan is returned
+				storedPlan, found := suite.chainA.GetSimApp().UpgradeKeeper.GetUpgradePlan(suite.chainA.GetContext())
+				suite.Require().True(found)
+				suite.Require().Equal(plan, storedPlan)
+
+				// check that old upgraded client state is cleared
+				_, found = suite.chainA.GetSimApp().UpgradeKeeper.GetUpgradedClient(suite.chainA.GetContext(), oldPlan.Height)
+				suite.Require().False(found)
+
+				// check that client state was set
+				storedClientState, found := suite.chainA.GetSimApp().UpgradeKeeper.GetUpgradedClient(suite.chainA.GetContext(), plan.Height)
+				suite.Require().True(found)
+				clientState, err := types.UnmarshalClientState(suite.chainA.App.AppCodec(), storedClientState)
+				suite.Require().NoError(err)
+				suite.Require().Equal(upgradedClientState, clientState)
+			} else {
+				suite.Require().True(errors.Is(err, expError))
+
+				// check that the new plan wasn't stored
+				storedPlan, found := suite.chainA.GetSimApp().UpgradeKeeper.GetUpgradePlan(suite.chainA.GetContext())
+				if oldPlan.Height != 0 {
+					// NOTE: this is only true if the ScheduleUpgrade function
+					// returns an error before clearing the old plan
+					suite.Require().True(found)
+					suite.Require().Equal(oldPlan, storedPlan)
+				} else {
+					suite.Require().False(found)
+					suite.Require().Empty(storedPlan)
+				}
+
+				// check that client state was not set
+				_, found = suite.chainA.GetSimApp().UpgradeKeeper.GetUpgradedClient(suite.chainA.GetContext(), plan.Height)
```

Comment:
nit: I have a preference for not discarding args as much as possible. We can do an assert empty or assert default struct of this value I think, right?